### PR TITLE
chore: use fstrings

### PR DESCRIPTION
**Pull request by branchvincent**
Towards https://github.com/python-poetry/poetry/issues/4776

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: poetry/core/json/__init__.py [hunk 1]
```diff
@@ -31,9 +31,7 @@ def validate_object(obj: dict, schema_name: str) -> List[str]:
     for error in validation_errors:
         message = error.message
         if error.path:
-            message = "[{}] {}".format(
-                ".".join(str(x) for x in error.absolute_path), message
-            )
+            message = f"[{'.'.join(str(x) for x in error.absolute_path)}] {message}"
```
Comment (danieleades) [inline poetry/core/json/__init__.py:34]: i think this is less clear than the original. consider splitting into two lines
```suggestion
            path = '.'.join(str(x) for x in error.absolute_path)
            message = f"[{path}] {message}"
```

File: poetry/core/masonry/builders/sdist.py [hunk 2]
```diff
@@ -123,7 +121,7 @@ def build_setup(self) -> bytes:
         # If we have a build script, use it
         if self._package.build_script:
             after += [
-                "from {} import *".format(self._package.build_script.split(".")[0]),
+                f"from {self._package.build_script.split('.')[0]} import *",
```
Comment (danieleades) [inline poetry/core/masonry/builders/sdist.py:124]: as above- this is harder to read than the original

File: poetry/core/packages/constraints/constraint.py [hunk 3]
```diff
@@ -126,6 +126,4 @@ def __hash__(self) -> int:
         return hash((self._operator, self._version))
 
     def __str__(self) -> str:
-        return "{}{}".format(
-            self._operator if self._operator != "==" else "", self._version
-        )
+        return f"{self._operator if self._operator != '==' else ''}{self._version}"
```
Comment (danieleades) [inline poetry/core/packages/constraints/constraint.py:129]: as above

File: poetry/core/packages/dependency_group.py [hunk 4]
```diff
@@ -49,6 +49,4 @@ def __eq__(self, other: object) -> bool:
         )
 
     def __repr__(self) -> str:
-        return "{}({}, optional={})".format(
-            self.__class__.__name__, self._name, self._optional
-        )
+        return f"{self.__class__.__name__}({self._name}, optional={self._optional})"
```
Comment (danieleades) [inline poetry/core/packages/dependency_group.py:52]: as above

**Review by danieleades — CHANGES_REQUESTED** [review]
there's quite a few places where the f-strings are *harder* to read than the original syntax. This occurs when the 'structure' of the original message is obscured by the long, inline code.

I commented on the first few, but there's a lot more places here where i think this holds.

In those cases, I would suggest improving clarity by adding intermediate calculations of the message components, and then using the local variables.

the point of this lint is to make things easier to grok, so we need to be careful about cargo-culting in the lint, and missing the point of why we're using it in the first place

**Comment by branchvincent** [comment]
thanks @danieleades, hopefully this is more readable

**Review by danieleades — APPROVED** [review]
Looks good to me!

File: src/poetry/core/packages/utils/link.py [hunk 5]
```diff
@@ -49,7 +49,7 @@ def __str__(self) -> str:
             return str(self.url)
 
     def __repr__(self) -> str:
-        return "<Link %s>" % self
+        return f"<Link {self}>"
```
Comment (neersighted) [inline src/poetry/core/packages/utils/link.py:52]:
```suggestion
        return f"<Link {self!s}>"
```

To keep this code consistent (`%s` implies the same `__str__` but not `__format__` call).

**Review by neersighted — CHANGES_REQUESTED** [review]
LGTM after one tiny correctness fix! Awesome work!

**Review by neersighted — APPROVED** [review]

**Pull request merged**